2.8.1 fails with "Cannot find module 'react-test-renderer/shallow'"

Downgrading to 2.8.0 fixes it.

So that prints pass, but when running the unit tests through karma, I get this:

```
13 04 2017 16:56:10.341:ERROR [framework.browserify]: bundle error
13 04 2017 16:56:10.346:ERROR [framework.browserify]: Error: Cannot find module 'react-dom/test-utils' from '<redacted>/node_modules/enzyme/build'
13 04 2017 16:56:10.365:INFO [karma]: Karma v0.13.22 server started at http://localhost:9876/
13 04 2017 16:56:10.373:INFO [launcher]: Starting browser PhantomJS
13 04 2017 16:56:10.430:ERROR [framework.browserify]: bundle error
13 04 2017 16:56:10.430:ERROR [framework.browserify]: Error: Cannot find module 'react-test-renderer/shallow' from '<redacted>/node_modules/enzyme/build'
```
`npm ls | grep react` shows me this:

```
├── @kadira/react-storybook-addon-info@3.3.0 extraneous
├─┬ babel-preset-react@6.24.1
│ ├─┬ babel-plugin-transform-react-display-name@6.23.0
│ ├─┬ babel-plugin-transform-react-jsx@6.24.1
│ │ ├─┬ babel-helper-builder-react-jsx@6.24.1
│ ├─┬ babel-plugin-transform-react-jsx-self@6.22.0
│ ├─┬ babel-plugin-transform-react-jsx-source@6.22.0
│ └─┬ react-router@2.8.1
├─┬ eslint-plugin-react@6.10.3
npm ├─┬ react@0.14.8
ERR!├── react-addons-test-utils@0.14.8
├── react-dom@0.14.8
├─┬ react-redux@5.0.4
│ ├─┬ create-react-class@15.5.2
│ ├── hoist-non-react-statics@1.2.0
├─┬ react-router@2.3.0
```

--- a/src/ShallowWrapper.js
+++ b/src/ShallowWrapper.js
@@ -102,7 +102,7 @@ function validateOptions(options) {
 
 function performBatchedUpdates(wrapper, fn) {
   const renderer = wrapper.root.renderer;
-  if (REACT155) {
+  if (REACT155 && renderer.unstable_batchedUpdates) {
     // React 15.5+ exposes batching on shallow renderer itself
     return renderer.unstable_batchedUpdates(fn);
   }

--- a/src/react-compat.js
+++ b/src/react-compat.js
@@ -96,19 +96,30 @@ if (REACT013) {
   // to list this as a dependency in package.json and have 0.13 work properly.
   // As a result, right now this is basically an implicit dependency.
   try {
-    if (REACT155) {
+    try {
+      // This is for react v15.5 and up...
+
       // eslint-disable-next-line import/no-extraneous-dependencies
       TestUtils = require('react-dom/test-utils');
-    } else {
+      // eslint-disable-next-line import/no-extraneous-dependencies
+      shallowRendererFactory = require('react-test-renderer/shallow').createRenderer;
+    } catch (e) {
+      // This is for react < v15.5.  Note that users who have `react^15.4.x` in their package.json
+      // will arrive here, too.  They need to upgrade.  React will print a nice warning letting
+      // them know they need to upgrade, though, so we're good.  Also note we explicitly do not
+      // use TestUtils from react-dom/test-utils here, mainly so the user still gets a warning for
+      // requiring 'react-addons-test-utils', which lets them know there's action required.
+
       // eslint-disable-next-line import/no-extraneous-dependencies
       TestUtils = require('react-addons-test-utils');
+      shallowRendererFactory = TestUtils.createRenderer;
     }
   } catch (e) {
     if (REACT155) {
       console.error( // eslint-disable-line no-console
-        'react-dom@15.5+ is an implicit dependency when using react@15.5+ with enzyme. ' +
-        'Please add the appropriate version to your devDependencies. ' +
-        'See https://github.com/airbnb/enzyme#installation',
+        'react-dom@15.5+ and react-test-renderer are implicit dependencies when using' +
+        'react@15.5+ with enzyme. Please add the appropriate version to your' +
+        'devDependencies. See https://github.com/airbnb/enzyme#installation',
       );
     } else {
       console.error( // eslint-disable-line no-console
@@ -117,26 +128,6 @@ if (REACT013) {
         'See https://github.com/airbnb/enzyme#installation',
       );
     }
-    throw e;
-  }
-
-  // Shallow renderer is accessible via the react-test-renderer package for React 15.5+.
-  // This is a separate package though and may not be installed.
-  try {
-    if (REACT155) {
-      // eslint-disable-next-line import/no-extraneous-dependencies
-      shallowRendererFactory = require('react-test-renderer/shallow').createRenderer;
-    } else {
-      // eslint-disable-next-line import/no-extraneous-dependencies
-      shallowRendererFactory = TestUtils.createRenderer;
-    }
-  } catch (e) {
-    // eslint-disable-next-line no-console
-    console.error(
-      'react-test-renderer is an implicit dependency in order to support react@15.5+. ' +
-      'Please add the appropriate version to your devDependencies. ' +
-      'See https://github.com/airbnb/enzyme#installation',
-    );
     throw e;
   }
 

